x-pack/legacy/plugins/infra/public/components/nodes_overview/table.tsx

```diff
@@ -42,126 +37,132 @@ const getGroupPaths = (path: SnapshotNodePath[]) => {
   }
 };
 
-export const TableView = class extends React.PureComponent<Props, State> {
-  public readonly state: State = initialState;
-  public render() {
-    const { nodes, options, formatter, currentTime, nodeType } = this.props;
-    const columns: Array<EuiBasicTableColumn<typeof items[number]>> = [
-      {
-        field: 'name',
-        name: i18n.translate('xpack.infra.tableView.columnName.name', { defaultMessage: 'Name' }),
-        sortable: true,
-        truncateText: true,
-        textOnly: true,
-        render: (value: string, item: { node: InfraWaffleMapNode }) => {
-          const tooltipText = item.node.id === value ? `${value}` : `${value} (${item.node.id})`;
-          // For the table we need to create a UniqueID that takes into to account the groupings
-          // as well as the node name. There is the possibility that a node can be present in two
-          // different groups and be on the screen at the same time.
-          const uniqueID = [...item.node.path.map(p => p.value), item.node.name].join(':');
-          return (
-            <NodeContextMenu
-              node={item.node}
-              nodeType={nodeType}
-              closePopover={this.closePopoverFor(uniqueID)}
-              currentTime={currentTime}
-              isPopoverOpen={this.state.isPopoverOpen.includes(uniqueID)}
-              options={options}
-              popoverPosition="rightCenter"
-            >
-              <EuiToolTip content={tooltipText}>
-                <EuiButtonEmpty onClick={this.openPopoverFor(uniqueID)}>{value}</EuiButtonEmpty>
-              </EuiToolTip>
-            </NodeContextMenu>
-          );
-        },
-      },
-      ...options.groupBy.map((grouping, index) => ({
-        field: `group_${index}`,
-        name: fieldToName((grouping && grouping.field) || ''),
-        sortable: true,
-        truncateText: true,
-        textOnly: true,
-        render: (value: string) => {
-          const handleClick = () => this.props.onFilter(`${grouping.field}:"${value}"`);
-          return (
-            <EuiToolTip content="Set Filter">
-              <EuiButtonEmpty onClick={handleClick}>{value}</EuiButtonEmpty>
+export const TableView = (props: Props) => {
+  const { nodes, options, formatter, currentTime, nodeType } = props;
+  const [isPopoverOpen, setIsPopoverOpen] = useState<string[]>([]);
```

Comment:
[nit] I see this was the previous name, but I'd rename this to `openPopovers` or similar. My first reaction with the current name is that: 
- There's only one popover
- The variable is a `boolean`